src/lattice/lattice_bravais.js

```diff
@@ -122,6 +122,155 @@ export class LatticeBravais {
         return object;
     }
 
+    /**
+     * Return lattice type from vectors array.
+     * @param {Number[][]} array
+     * @returns {String}
+     */
+    // eslint-disable-next-line no-unused-vars
+    static typeFromVectors(array) {
+        const type = "TRI"; // TODO: remove temporary assigment corresponding to default value in basis constructor
+        // TODO: implement
+        return type;
+    }
+
+    /**
+     * Return primitive lattice vectors from type and necessary parameters of: a, b, c, cosbc, cosac, cosab.
+     * According to this: https://arxiv.org/pdf/1004.2974.pdf
+     * @param {Object} config
+     * @param {String} config.type
+     * @param {Number} config.a
+     * @param {Number} config.b
+     * @param {Number} config.c
+     * @param {Number} config.alpha - angle between b and c in degrees
+     * @param {Number} config.beta - angle between a and c in degrees
+     * @param {Number} config.gamma - angle between a and b in degrees
+     * @returns {{vectors: Number[][], type: String}}
+     */
+    // eslint-disable-next-line no-unused-vars
+    static vectorsFromType({ type, a = 1, b = a, c = a, alpha = 90, beta = 90, gamma = 90 }) {
```

Comment:
Remove. This is already done in `fromBravais`, see `PRIMITIVE_CELLS`

Reply:
Yes. I've noticed that